### ISSUE TITLE
v1alpha2 objects do not get vetted by v1alpha3 validation webhooks

### ISSUE THREAD

**Issue by sedefsavas** [kubernetes-sigs/cluster-api#2513]
**What steps did you take and what happened:**
Applied the following v1alpha2 Cluster yaml which has inconsistency between infrastructureRef namespace and Cluster namespace. This inconsistency should have been catched by v1alpha3 Cluster validation webhooks, but instead the Cluster object is being created without any errors. This is the case for other resources  too (such as Machine).
```
apiVersion: cluster.x-k8s.io/v1alpha2
kind: Cluster
metadata:
  name: management-cluster
  namespace: default
spec:
  clusterNetwork:
    pods:
      cidrBlocks:
      - 100.96.0.0/11
    serviceDomain: cluster.local
    services:
      cidrBlocks:
      - 100.64.0.0/13
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
    kind: VSphereCluster
    name: management-cluster
    namespace: different-ns
```

**What did you expect to happen:**
I expected to see the same error I get from applying the exact yaml file by only changing v1alpha2 to v1alpha3, which is:

```
Error from server (Cluster.cluster.x-k8s.io "management-clusterx" is invalid: spec.infrastructureRef.namespace: Invalid value: "different-ns": must match metadata.namespace): error when creating "v1alpha2cluster.yml": admission webhook "validation.cluster.cluster.x-k8s.io" denied the request: Cluster.cluster.x-k8s.io "management-clusterx" is invalid: spec.infrastructureRef.namespace: Invalid value: "different-ns": must match metadata.namespace
```
Related issue: https://github.com/kubernetes-sigs/cluster-api/issues/2437

**Environment:**

- Cluster-api version: v0.3.0-rc.2

/kind bug

**Comment by ncdc** [comment]
This should get fixed when we do #2442

**Comment by vincepri** [comment]
Closing in favor of https://github.com/kubernetes-sigs/cluster-api/issues/2442

/close
